### PR TITLE
Add support for MOID on datacenter field

### PR DESCRIPTION
**Pull request by rikatz**
**What this PR does / why we need it**: This PR adds support for MOID on the datacenter field, so the real object ID can be used instead of its canonical name

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Support for MOID on field names
```

### CODE REVIEW

File: pkg/common/vclib/datacenter.go [hunk 1]
```diff
@@ -40,12 +40,27 @@ type Datacenter struct {
...
 func GetDatacenter(ctx context.Context, connection *VSphereConnection, datacenterPath string) (*Datacenter, error) {
-	finder := find.NewFinder(connection.Client, false)
-	datacenter, err := finder.Datacenter(ctx, datacenterPath)
-	if err != nil {
-		klog.Errorf("Failed to find the datacenter: %s. err: %+v", datacenterPath, err)
-		return nil, err
+	var datacenter *object.Datacenter
+	var err error
+
+	dcRef := object.ReferenceFromString(datacenterPath)
```
Comment (lubronzhan) [inline pkg/common/vclib/datacenter.go:46]: The name of the function is not that explicit. Could you add a comment to say check if the datacenter path is moid format or not?

**Review by lubronzhan — APPROVED** [review]
Overall looks good to me.

File: pkg/common/vclib/datacenter.go (hunk 1 above)
Comment (rikatz) [reply pkg/common/vclib/datacenter.go:49]: done

**Comment by lubronzhan** [comment]
cc @DanielXiao @XudongLiuHarold this is for vanila k8s cluster

/approve
/lgtm

**Pull request merged**